### PR TITLE
Add pattern of test file and update pull_request_template

### PR DESCRIPTION
**Pull request by tdrozdovsky**
Signed-off-by: Taras Drozdovskyi <t.drozdovsky@samsung.com>

# Description

Added a template for creating a test file and updated the pull_request_template.

## Type of change

Please delete options that are not relevant.

- [x] Documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: docs/testing_policy.md [hunk 1]
```diff
@@ -53,4 +54,51 @@ go test -v [PKG_NAME]
...
+	})
+    // Negative tests
+	t.Run("Fail", func(t *testing.T) {
+        // Nested test
+		t.Run("FailCondition #1", func(t *testing.T) {
+			if err := FuncName(param); err == nil {
+				t.Error(unexpectedSuccess)
+			}
+		})
+        // Nested test with panic result
```
Comment (t25kim) [inline docs/testing_policy.md:91]: I would like to modify the indentation as below.

```suggestion
	// Positive test
	t.Run("Success", func(t *testing.T) {
		if err := FuncName(param); err != nil {
			t.Error(unexpectedFail)
		}
	})
	// Negative tests
	t.Run("Fail", func(t *testing.T) {
        	// Nested test
		t.Run("FailCondition #1", func(t *testing.T) {
			if err := FuncName(param); err == nil {
				t.Error(unexpectedSuccess)
			}
		})
		// Nested test with panic result
```

Comment (tdrozdovsky) [reply docs/testing_policy.md]: Done! )

**Review by t25kim — APPROVED** [review]
LGTM Thanks!

**Review by MoonkiHong — APPROVED** [review]
Looks good to me. 😁

**Pull request merged**